backend/app/controllers/youtube_controller.rb

```
@@ -0,0 +1,230 @@
+          video_url: "https://www.youtube.com/watch?v=#{item.content_details.video_id}",
+          position: item.snippet.position
+        }
+      end
+
+      all_tracks.concat(tracks)
+      next_page_token = response.next_page_token
+      break unless next_page_token
+    end
+
+    all_tracks.sort_by { |track| track[:position] }
+  end
```

RuboCop (1.76.1)

[convention] Assignment Branch Condition size for `fetch_playlist_tracks` is too high. [<8, 24, 7> 26.25/23]

(Metrics/AbcSize)